src/game_map.cpp

```diff
@@ -1329,17 +1329,13 @@ void Game_Map::UpdateParallax() {
 		return;
 
 	if (map_info.parallax_horz && map_info.parallax_horz_auto) {
-		parallax_x +=
-			map_info.parallax_horz_speed > 0 ? 1 << (map_info.parallax_horz_speed - 1) :
-			map_info.parallax_horz_speed < 0 ? -1 << (-map_info.parallax_horz_speed - 1) :
-			0;
+		int acc = 1 << (std::abs(map_info.parallax_horz_speed) - 1);
```

Comment:
If `map_info.parallax_horz_speed` is zero, the RHS of the left shift is negative. I don't know whether this case occurs or not, but the old code handles it.